File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 2/11/2026, 11:08:49 AM
class Solution {
    public boolean isPalindrome(int x) {
        if (x<0){
            return false;
        }
        StringBuilder z = new StringBuilder(String.valueOf(x));
        if (z.toString().equals(z.reverse().toString())){
            return true;
        }
        return false;
    }
}

File: 9-PalindromeNumber/9-PalindromeNumber.scala
// Last updated: 2/11/2026, 11:08:46 AM
object Solution {
    def isPalindrome(x: Int): Boolean = {
        if (x < 0 || (x % 10 == 0 && x != 0)) return false

        var num = x
        var reversed = 0

        while (num > reversed) {
        reversed = reversed * 10 + num % 10
        num /= 10
        }

        num == reversed || num == reversed / 10
    }
}